lib/modules/hosts/giphy.js

```diff
@@ -1,27 +1,35 @@
+import { ajax } from '../../environment';
+import logo from '../../images/hosts/giphy-logo.png';
+
 export default {
 	moduleID: 'giphy',
 	name: 'giphy',
 	domains: ['giphy.com'],
-	logo: '//giphy.com/static/img/favicon.png',
-	detect: ({ pathname }) => (/^\/gifs\/(?:\w+-)*(.*?)(\/html5)?$/i).exec(pathname),
-	handleLink(href, [, id, isHtml5]) {
+	logo,
+	detect: ({ pathname }) => (/^(?:\/gifs|\/media|)\/(?:\w+-)*([^/.]*)(\/html5)?.*$/i).exec(pathname),
+	async handleLink(href, [, id, isHtml5]) {
+		const { data } = await ajax({
+			url: `http://api.giphy.com/v1/gifs/${id}?api_key=dc6zaTOxFJmzC`,
+			type: 'json',
+		});
+
 		if (isHtml5) {
```

Comment:
This condition can be removed completely (and always take the first branch).

It seems that that giphy API always returns an mp4, and if it doesn't we'll just fall back to the same `.gif`.